jib-cli/src/test/java/com/google/cloud/tools/jib/cli/buildfile/BuildFilesTest.java

```diff
@@ -0,0 +1,108 @@
+import com.google.cloud.tools.jib.api.buildplan.Platform;
+import com.google.cloud.tools.jib.api.buildplan.Port;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.Resources;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Instant;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class BuildFilesTest {
+
+  @Rule public final TemporaryFolder tmp = new TemporaryFolder();
+
+  @Test
+  public void testToJibContainerBuilder_allProperties()
+      throws URISyntaxException, IOException, InvalidImageReferenceException {
+    URL resource = Resources.getResource("buildfiles/projects/allProperties/jib.yaml");
+    JibContainerBuilder jibContainerBuilder =
+        BuildFiles.toJibContainerBuilder(Paths.get(resource.toURI()));
+    Path projectRoot = Paths.get(resource.toURI()).getParent();
+
+    ContainerBuildPlan resolved = jibContainerBuilder.toContainerBuildPlan();
+    Assert.assertEquals("ubuntu", resolved.getBaseImage());
+    Assert.assertEquals(Instant.ofEpochMilli(2000), resolved.getCreationTime());
+    Assert.assertEquals(ImageFormat.OCI, resolved.getFormat());
+    Assert.assertEquals(
+        ImmutableSet.of(new Platform("arm", "linux"), new Platform("amd64", "darwin")),
+        resolved.getPlatforms());
+    Assert.assertEquals(ImmutableMap.of("KEY1", "v1", "KEY2", "v2"), resolved.getEnvironment());
+    Assert.assertEquals(
+        ImmutableSet.of(AbsoluteUnixPath.get("/volume1"), AbsoluteUnixPath.get("/volume2")),
+        resolved.getVolumes());
+    Assert.assertEquals(ImmutableMap.of("label1", "l1", "label2", "l2"), resolved.getLabels());
+    Assert.assertEquals(
+        ImmutableSet.of(Port.udp(123), Port.tcp(456), Port.tcp(789)), resolved.getExposedPorts());
+    Assert.assertEquals("customUser", resolved.getUser());
+    Assert.assertEquals(AbsoluteUnixPath.get("/home"), resolved.getWorkingDirectory());
+    Assert.assertEquals(ImmutableList.of("sh", "script.sh"), resolved.getEntrypoint());
+    Assert.assertEquals(ImmutableList.of("--param", "param"), resolved.getCmd());
+
+    Assert.assertEquals(1, resolved.getLayers().size());
+    FileEntriesLayer resolvedLayer = (FileEntriesLayer) resolved.getLayers().get(0);
+    Assert.assertEquals("scripts", resolvedLayer.getName());
+    Assert.assertEquals(
+        FileEntriesLayer.builder()
+            .addEntry(
+                projectRoot.resolve("project/script.sh"), AbsoluteUnixPath.get("/home/script.sh"))
+            .build()
+            .getEntries(),
+        resolvedLayer.getEntries());
+    Assert.assertEquals(LayerObject.Type.FILE_ENTRIES, resolvedLayer.getType());
+  }
+
+  @Test
+  public void testToJibContainerBuilder_requiredProperties()
```

Comment:
quick question on this: Is this test only for when a scratch base image is used? And is that why we allow some of these required parameters to be empty or null?

Reply:
This test adds in only the required properties and then verifies that the output is the "defaults". When the defaults change, this will fail and we will know that we either have to revert the change or update our documentation to correctly reflect what the defaults are.

Reply:
Cool, thanks!